src/Plugin/MediaEntity/Type/EmbeddableVideo.php

```diff
@@ -205,12 +202,24 @@ public function settingsForm(MediaBundleInterface $bundle) {
   /**
    * {@inheritdoc}
    */
-  public function validate(MediaInterface $media) {
-    if ($this->matchProvider($media)) {
-      return;
+  public function attachConstraints(MediaInterface $media) {
+    parent::attachConstraints($media);
+
+    $source_field_name = $this->configuration['source_field'];
+
+    // Get all providers regexes. Wehen we will be able to select providers
+    // per field we should handle that here.
+    $video_provider_definitions = $this->videoProviders->getDefinitions();
+    $regexes = [];
+    foreach ($video_provider_definitions as $definition) {
+      $regexes = array_merge($regexes, $definition['regular_expressions']);
     }
 
-    throw new MediaTypeException($this->configuration['source_field'], 'Not valid URL/embed code.');
+    foreach ($media->get($source_field_name) as &$embed_code) {
+      /** @var \Drupal\Core\TypedData\DataDefinitionInterface $typed_data */
+      $typed_data = $embed_code->getDataDefinition();
+      $typed_data->addConstraint('VideoProviderMatch', ['regular_expressions' => $regexes]);
```

Comment:
Since this checks any embed code I'm thinking if we should make this generic constraint. All providers that are based of embed codes could use it then.